src/libraries/System.Net.Security/src/System/Net/Security/SecureChannel.cs

```diff
@@ -818,7 +818,7 @@ private SecurityStatusPal GenerateToken(ReadOnlySpan<byte> inputBuffer, ref byte
                     // This call may bump up the credential reference count further.
                     // Note that thumbPrint is retrieved from a safe cert object that was possible cloned from the user passed cert.
                     //
-                    if (!cachedCreds && _securityContext != null && !_securityContext.IsInvalid && _credentialsHandle != null && !_credentialsHandle.IsInvalid)
+                    if (!cachedCreds && status.ErrorCode == SecurityStatusPalErrorCode.OK && _securityContext != null && !_securityContext.IsInvalid && _credentialsHandle != null && !_credentialsHandle.IsInvalid)
```

Comment:
Do we have tests that will end up with a non-OK ErrorCode?

Reply:
I will take a look. There seems to be some race condition so I see the "Close handle" only once in a while. It would be _nice_ to have some predictability for error cases.